pkg/awsutils/awsutils.go

```diff
@@ -315,8 +331,27 @@ func (i instrumentedIMDS) GetMetadataWithContext(ctx context.Context, p string)
 	return result, nil
 }
 
+func (cache *EC2InstanceMetadataCache) GetNetworkInterfacesWithContext(ctx context.Context, eniIds []*string) (*ec2.DescribeNetworkInterfacesOutput, error) {
+	input := &ec2.DescribeNetworkInterfacesInput{NetworkInterfaceIds: eniIds}
+
+	start := time.Now()
+	result, err := cache.ec2SVC.DescribeNetworkInterfacesWithContext(context.Background(), input)
```

Comment:
seems this API is paginated, is it handled within ec2SVC?

Reply:
Pagination is done only for this func - https://github.com/aws/amazon-vpc-cni-k8s/blob/9ad4860bbe2f7fe8548670447ac69c8ee989d872/pkg/awsutils/awsutils.go#L1326 since we do describe on all ENIs, the calls here does describe on only certain ENIs.

Reply:
if it accepts a list, we need to handle pagination correctly within that function. or change it to accept a single eni.

Reply:
Fixed the API.